apps/cowswap-frontend/src/modules/swap/services/swapFlow/index.ts

```diff
@@ -27,16 +27,18 @@ export async function swapFlow(
 
   try {
     logTradeFlow('SWAP FLOW', 'STEP 2: handle permit')
+    if (input.permitInfo) input.swapConfirmManager.requestPermitSignature()
+
     input.orderParams.appData = await handlePermit({
       appData: input.orderParams.appData,
-      hasEnoughAllowance: input.hasEnoughAllowance,
 
       inputToken: input.context.trade.inputAmount.currency as Token,
       provider: input.orderParams.signer.provider as Web3Provider,
       account: input.orderParams.account,
       chainId: input.orderParams.chainId,
       permitInfo: input.permitInfo,
     })
+    input.swapConfirmManager.permitSigned()
```

Comment:
Here's the other side of the magic.
Request signature when we get to this state to show the permit modal.
Move it to next stage after signed.